### PR TITLE
add kserve-rest-proxy package

### PR DESCRIPTION
**Pull request by shimmerjs**
Fixes:

Related:

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: kserve-rest-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  version: 0.12.0
+  epoch: 0
+  description: "REST API proxy for kserve, the standardized serverless ML inference platform on Kubernetes"
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - go
```
Comment (EyeCantCU) [inline kserve-rest-proxy.yaml:12]: We can drop the environment here. `go` is included as a part of `go/build` below

Comment (shimmerjs) [inline kserve-rest-proxy.yaml:12]:
```suggestion
```

File: kserve-rest-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: kserve-rest-proxy
+  version: 0.12.0
+  epoch: 0
+  description: "REST API proxy for kserve, the standardized serverless ML inference platform on Kubernetes"
+  copyright:
+    - license: Apache-2.0
+
+
```
Comment (shimmerjs) [inline kserve-rest-proxy.yaml:9]:
```suggestion
```

File: kserve-rest-proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    with:
+      modroot: .
+      packages: ./proxy
+      output: rest-proxy
+
+test:
+  environment:
+    contents:
+      packages:
+        - curl
```
Comment (smoser) [inline kserve-rest-proxy.yaml:26]: you only need curl if you're going to check it. which is good to do , but you're not doing that right now.

File: kserve-rest-proxy.yaml (hunk 1 above)
Comment (shimmerjs) [reply kserve-rest-proxy.yaml]: TY TY, addressed

File: kserve-rest-proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (shimmerjs) [inline kserve-rest-proxy.yaml:26]:
```suggestion
```

File: kserve-rest-proxy.yaml (hunk 3 above)
Comment (shimmerjs) [reply kserve-rest-proxy.yaml]: TY, missed removing that once I realized health checking wouldn't be as trivial as a `curl` request.

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**